apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.component.ts

```diff
@@ -2,20 +2,30 @@ import { ChangeDetectionStrategy, Component, computed, effect, type OnDestroy, t
 import { toSignal } from '@angular/core/rxjs-interop';
 import { FormControl, FormGroup, FormsModule, ReactiveFormsModule } from '@angular/forms';
 import { DfTooltipModule } from '@design-factory/design-factory';
-import { NgbTypeaheadModule } from '@ng-bootstrap/ng-bootstrap';
-import { type GetStylingVariableContentMessage, type StylingVariable, THEME_TAG_NAME } from '@o3r/styling';
+import { NgbAccordionModule, NgbTypeaheadModule } from '@ng-bootstrap/ng-bootstrap';
+import { computeItemIdentifier } from '@o3r/core';
+import { type GetStylingVariableContentMessage, PALETTE_TAG_NAME, type StylingVariable, THEME_TAG_NAME } from '@o3r/styling';
 import { combineLatest, Observable, Subscription } from 'rxjs';
 import { filter, map, shareReplay, startWith, throttleTime } from 'rxjs/operators';
 import { ChromeExtensionConnectionService } from '../../services/connection.service';
-import { ColorPipe } from './color.pipe';
-import { resolveVariable, searchFn } from './common';
+import { ColorPipe, ConstrastPipe } from './color.pipe';
+import { DEFAULT_VARIANT, getPaletteColors, getVariant, resolveVariable, searchFn } from './common';
 import { IsRefPipe } from './is-ref.pipe';
 import { MemoizePipe } from './memoize.pipe';
 import { VariableLabelPipe } from './variable-label.pipe';
 import { VariableNamePipe } from './variable-name.pipe';
 
 const THROTTLE_TIME = 100;
 
+type VariableGroupType = 'type' | 'category' | 'component';
+
+interface VariableGroup {
```

Comment:
It actually needs comments because it is used as parameter of a public function

Reply:
In that case, shouldn't it also be exported?